### DISCUSSION TITLE
Guided Exercise - HTML - Crafting a Multi-Feature Web Page Using Basic HTML

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#230] [Tasks]
This exercise will help reinforce fundamental HTML concepts while introducing some creative uses of these elements.

#### Objective:
Develop a web page that integrates various basic HTML elements in a creative manner, showcasing multiple uses of these tags.

#### Step 1: Prepare Your Workspace
- **Create a New HTML File:** Name it `creative_basics.html` in your `html_basics` project folder.

#### Step 2: Basic HTML Structure
- **Start with Skeleton:** Set up your document with the basic HTML5 skeleton (doctype, html, head, body).

#### Step 3: Crafting a Creative Header
- **Multiple Headings:** Use `<h1>` for the main title and `<h2>` to `<h6>` for subtitles, taglines, or quotes.
- **Horizontal Rule:** Use `<hr>` tags to create a thematic break between header sections.

#### Step 4: Structuring Content with Paragraphs and Lists
- **Paragraphs with Emphasis:** Use `<p>` for paragraphs. Within paragraphs, use `<strong>` for strong emphasis and `<em>` for italicized text.
- **Nested Lists:** Create a list (either `<ul>` or `<ol>`) and within one list item, nest another list.

#### Step 5: Linking and Quoting
- **Creative Links:** Use the `<a>` tag to link to external sites, but also to create a "Table of Contents" at the top that links to different sections of your page using anchors (`<a href="#section-id">`).
- **Blockquote:** Use the `<blockquote>` tag for quoting external sources, followed by a citation using `<cite>`.

#### Step 6: Image Gallery
- **Multiple Images:** Use the `<img>` tag to create a small image gallery. Use the `alt` attribute for descriptions.
- **Captions:** Below each image, use `<figcaption>` to provide a caption.

#### Step 7: Implementing a Mini Contact Form
- **Simple Form:** Use the `<form>` tag to create a basic form. Include:
  - **Input Fields:** Use `<input type="text">` for name and email fields.
  - **Submit Button:** Add a submit button using `<input type="submit">`.

#### Step 8: Creative Footer
- **Footer Content:** Use the `<footer>` tag and include:
  - **Copyright Notice:** Use a paragraph tag.
  - **Social Media Links:** Use `<a>` tags to simulate links to social media.

#### Step 9: View and Test Your Web Page
- **Browser Preview:** Open `creative_basics.html` in a web browser and see how your creative use of basic HTML elements looks.

#### Step 10: Reflect on Your Learning
- **Document Your Process:** Write a brief summary of how you used basic HTML tags in various ways.
- **Challenges and Solutions:** Note any challenges you faced and how you resolved them.

---

This exercise encourages creativity within the constraints of basic HTML, allowing learners to explore the versatility of foundational elements. By engaging with HTML tags in multiple contexts, learners can gain a deeper appreciation of how these elements come together to form the structure of a web page.

---

Let's break down the exercise into detailed implementation steps, providing clear instructions and explanations for each part. This will guide learners through creating a multi-featured web page using basic HTML elements creatively.

#### Step 1: Prepare Your Workspace
1. **Create a New HTML File:** Open your code editor and create a new file named `creative_basics.html` in your `html_basics` project folder.

#### Step 2: Basic HTML Structure
1. **Start with Skeleton:** Write the basic HTML structure:
   ```html
   <!DOCTYPE html>
   <html>
   <head>
       <title>Creative HTML Basics</title>
   </head>
   <body>
       <!-- Content will go here -->
   </body>
   </html>
   ```
   - **Explanation:** `<!DOCTYPE html>` declares the document type and HTML version. The `<html>` tag is the root of the HTML document. The `<head>` section contains meta-information like the `<title>`.

#### Step 3: Crafting a Creative Header
1. **Add Main Title:** Use `<h1>` for your main page title.
2. **Add Subtitles:** Include `<h2>` to `<h6>` tags for additional headings or thematic breaks.
3. **Horizontal Rule:** Insert `<hr>` tags to visually separate sections in the header.
   - **Explanation:** Headings (`<h1>` to `<h6>`) are used for creating a hierarchical structure. `<hr>` creates a thematic break in the page, often used for styling.

#### Step 4: Structuring Content with Paragraphs and Lists
1. **Paragraphs:** Write a few paragraphs using the `<p>` tag.
2. **Emphasis:** Within these paragraphs, use `<strong>` and `<em>` to emphasize text.
3. **Nested Lists:** Create an ordered (`<ol>`) or unordered (`<ul>`) list. Inside one of the list items, nest another list.
   - **Explanation:** Paragraphs are fundamental for text content. `<strong>` indicates strong importance, while `<em>` denotes emphasized text. Nested lists show hierarchy or categorization.

#### Step 5: Linking and Quoting
1. **Create Internal Links:** Use `<a href="#section-id">` to link to sections on your page.
2. **Add a Blockquote:** Use `<blockquote>` for quoting, and `<cite>` for citations.
   - **Explanation:** Anchor tags (`<a>`) are used for linking. The `href` attribute specifies the link’s destination. `<blockquote>` is used for extended quotations, and `<cite>` for citations.

#### Step 6: Image Gallery
1. **Insert Images:** Use `<img>` to add images, with `alt` attributes for accessibility.
2. **Add Captions:** Below each image, use `<figcaption>` to provide a caption.
   - **Explanation:** The `<img>` tag embeds images, while `alt` text describes the image. `<figcaption>` offers a caption or explanation for the image.

#### Step 7: Implementing a Mini Contact Form
1. **Form Tag:** Use `<form>` to create a form container.
2. **Input Fields:** Add `<input type="text">` for name and email.
3. **Submit Button:** Include `<input type="submit">` for the form submission.
   - **Explanation:** The `<form>` tag is used for collecting user input. `<input type="text">` creates text fields, and `<input type="submit">` creates a button to submit the form.

#### Step 8: Creative Footer
1. **Footer Section:** Use `<footer>` for the bottom part of your page.
2. **Content:** Add a copyright notice using `<p>`. Simulate social media links with `<a>` tags.
   - **Explanation:** The `<footer>` element is used for containing footer content, typically found at the bottom of the page.

#### Step 9: View and Test Your Web Page
1. **Open in Browser:** Save your `creative_basics.html` file and open it in a web browser to see the results.

#### Step 10: Reflect on Your Learning
1. **Document the Process:** Write notes about how you used each HTML tag and attribute.
2. **Identify Challenges:** Reflect on any difficulties faced and how you solved them.

---

This step-by-step exercise guides learners through the process of building a web page using basic HTML elements in a creative way. It's designed to reinforce fundamental concepts while encouraging experimentation with the structure and presentation of content.

**Comment by Jamal-bacchus** [comment]
<img width="565" alt="Screenshot 2023-11-18 at 11 46 06 AM" src="https://github.com/akash-coded/mern/assets/92058171/21df87cc-b365-4535-8408-f931bfe9d99f">
done

**Comment by sidchakravarty** [comment]
![Assignment - 2](https://github.com/akash-coded/mern/assets/15659742/b718643d-d34a-4f1d-a324-2194d63d47a6)

**Reply by priyeranjan** [reply]
<img width="955" alt="image" src="https://github.com/akash-coded/mern/assets/7481567/f93a00ee-7835-486c-a966-5475809c3495">

**Comment by monalip92** [comment]
![Uploading capturepage.png…]()

**Comment by dmpatil1990** [comment]
![Uploading GE-230.png…]()

**Comment by monalip92** [comment]
![capturepage](https://github.com/akash-coded/mern/assets/144552652/6f748296-19ce-4ea1-a03d-7a13b802331a)

**Comment by vpervela** [comment]
<img width="1191" alt="Screen Shot 2023-11-18 at 9 32 32 AM" src="https://github.com/akash-coded/mern/assets/101532142/d4d2c62f-39b7-46a2-9ab9-b4b3a72f8a94">

**Comment by Dhanalakshmi-S-23** [comment]
![230](https://github.com/akash-coded/mern/assets/151384783/a022de07-ebb1-4505-8183-9dd27558b7f7)

**Comment by ghost** [comment]
![Screenshot (256)](https://github.com/akash-coded/mern/assets/135525365/75ed321d-cd70-42cc-9a0c-3738928f45fe)

**Comment by arvid-mvo** [comment]
![upload_2](https://github.com/akash-coded/mern/assets/101291172/22ec9973-f3b9-4081-89fb-aade9e254979)

**Comment by ujjwal432** [comment]
<img width="1078" alt="Html Exercise #230" src="https://github.com/akash-coded/mern/assets/61551049/64e21f2e-b2ac-4db1-a284-12bdf7510e08">

**Comment by Lellagopi143** [comment]
[
![Screenshot 2023-11-19 085957](https://github.com/akash-coded/mern/assets/151382935/c2cbdf24-d44f-4cb3-b209-e4ecc2948bc4)
![Screenshot 2023-11-19 090004](https://github.com/akash-coded/mern/assets/151382935/7ddb59a0-039c-4764-9abc-9398554ee803)
](url)

**Comment by AbiramiMadhusudan28** [comment]
<img width="1470" alt="230" src="https://github.com/akash-coded/mern/assets/151383196/19d663d5-4a44-4610-ba16-c978d8541c17">

**Comment by JJismi** [comment]
<img width="960" alt="h1" src="https://github.com/akash-coded/mern/assets/133745306/f0ba8c08-547c-464c-a1d5-51013dd43965">

**Comment by Dipalivrma** [comment]
![image](https://github.com/akash-coded/mern/assets/119152780/1e3977f4-5dd2-4a1d-a932-fae9dcd79a64)

**Comment by SwetaSaha1** [comment]
![Screenshot 2023-11-19 at 6 42 54 PM](https://github.com/akash-coded/mern/assets/127398967/7e2369dd-573f-4911-896a-a60b833df03c)

**Comment by santhoshraj123** [comment]
![IMG20231119184733.jpg](https://github.com/akash-coded/mern/assets/827096/b070cd72-8b9c-4c63-9469-005ca569870f)

**Comment by mygit-shirinA** [comment]
![image](https://github.com/akash-coded/mern/assets/151388915/202673ba-ed3a-4352-af8e-925cc1e31e41)
Submitted by Shirin (under Maria Ebrahimji)

**Comment by VarshiniBS** [comment]
![Screenshot (687)](https://github.com/akash-coded/mern/assets/87222601/131b0168-dc21-49eb-aebf-e86c1ff9fa9b)

**Comment by SwetaSaha1** [comment]
![Uploading Screenshot 2023-11-19 at 7.17.24 PM.png…]()

**Comment by SwetaSaha1** [comment]
![Screenshot 2023-11-19 at 7 17 24 PM](https://github.com/akash-coded/mern/assets/127398967/5cee786c-b2e5-4ac4-89e2-c8c9ecc7b2c3)

**Comment by padma1998lela** [comment]
![image](https://github.com/akash-coded/mern/assets/62737217/a386e7f2-1ad4-44a4-abbb-2cc06976a453)

**Comment by Umarani1711** [comment]
<img width="954" alt="Creative Basic1" src="https://github.com/akash-coded/mern/assets/78922312/44f0c525-a21c-4cc2-bb87-26aae978f538">
<img width="960" alt="Creative Basic2" src="https://github.com/akash-coded/mern/assets/78922312/55114896-ffa6-429a-bec8-0fec14ce503d">

**Comment by bacchus19408** [comment]
<img width="656" alt="Screen Shot 2023-11-22 at 7 34 03 AM" src="https://github.com/akash-coded/mern/assets/4041048/b641507a-78b6-44ae-a6c8-7945aa653fd7">

**Reply by bacchus19408** [reply]
i forgot to add the send button 
<img width="684" alt="Screen Shot 2023-11-22 at 3 28 21 PM" src="https://github.com/akash-coded/mern/assets/4041048/f7a7c9f6-a988-490b-9688-85e9fc6d0dab">

**Comment by Sunil-nani** [comment]
![Screenshot (7)](https://github.com/akash-coded/mern/assets/144610117/89719c22-9037-42f0-aef5-85f9a0848433)

**Comment by PUMAGrisen** [comment]
![Namnlös](https://github.com/akash-coded/mern/assets/151846056/625bd9a3-2879-440d-a649-81958c5bafde)

**Comment by Geethanjali5** [comment]
![IMG-20231123-WA0001](https://github.com/akash-coded/mern/assets/147892641/cb10929b-209c-4e82-bac2-493cc24dcb89)

**Comment by VIJAY1999v** [comment]
![Uploading Screenshot (329).png…]()
